### ISSUE TITLE
Notification Panel: Include Encrypted rooms (part 1)

### ISSUE THREAD

**Issue by daniellekirkwood** [vector-im/element-web#25383]
### Your use case

The notification panel in Element web currently only displays unencrypted ping/highlight messages (mentions and keywords). The panel is extremely useful to people and internally we had a discussion [here](https://matrix.to/#/!kxwQeJPhRigXSZrHqf:matrix.org/$yuFgU3b3Da6GVA795ktBSf-F_b5gCkFTjhUVJEq9Rb4?via=matrix.org&via=element.io&via=vector.modular.im)

In the future we want the notification panel to:
- Include all messages you were mentioned or keyword-ed in, regardless of encrypted or unencrypted
- Move the access point (it's wrong on so many levels today 😂 )

Adding encrypted rooms to this list may take some work so we'd like to experiment with ways we might be able to do this. This issue is _only_ for the first experiment we've discussed. 

---
**Can we add encrypted rooms to the notifications panel?**
- Yes, but performance will take a hit unless we make some changes to spec

Our suggestion: 
- Have the notifications panel only keep messages from the last 24 hours. This reduces the load and therefore shouldn't have too much of a negative impact on the product. (In the future we should aim to remove or lengthen this cap, but for the purposes of testing how we'd do this, it's a good compromise for now) 

---
**Open questions**

Things we're thinking about: 
- If we shorten the unencrypted rooms 'memory' in the panel to only 24 hours user's may get upset
- If we add the two lists together users may get confused

To be discussed:
- Could we have a drop down at the top of the panel (similar to the threads panel) that let's users choose between 'forever unencrypted' and '24 hour encrypted' while we're in this experimental phase?
- Or, could we have a labs flag and measure how many people turn it on and are willing to only see the last 24 hours in order to get all their rooms?
- Or, could we extend the 24 hours to longer (maybe 3 or 4 days) and what would the performance impact/change be?

Spec land:
- What changes would we have to make server-side to help the client provide this to users?

**Comment by daniellekirkwood** [comment]
@giomfo @weeman1337 

here's the issue we were talking about yesterday with my 'open questions' that i'd like us to discuss. 

WDYT?

**Comment by t3chguy** [comment]
Duplicate of https://github.com/vector-im/element-web/issues/6874

**Comment by daniellekirkwood** [comment]
it's not a duplicate.

**Comment by t3chguy** [comment]
The `/notifications` API should be sufficient for this. Currently we call it with `only=highlight` which makes it exclude encrypted notifications as by default those only trigger `notify` push actions and `highlight` only once decrypted.

We could drop the `only=highlight` and perform a local filtering.
Synapse drops record of non-highlight notifications after a day so it'd match the 24 hr POC. This isn't spec conforming, the spec says 

> This API is used to paginate through the list of events that the user has been, or would have been notified about.

It does not have any note which permits a server to prune this resultant set.

---

A local filtering isn't trivial, either we slow down the pagination of this panel to allow for decryption or we allow things to be in the list slightly out of order to make it feel faster

**Comment by giomfo** [comment]
Note: MSC4028 may help on this request (see [here](https://github.com/matrix-org/matrix-spec-proposals/blob/giomfo/push_encrypted_events/proposals/4028-push-all-encrypted-events-except-for-muted-rooms.md#listening-notifications))

**Comment by t3chguy** [comment]
@giomfo yes that will be needed to support M&K Encrypted rooms
